File: packages/js-sdk/src/regex/detectors/ro/cui.ts
// Romanian CUI (company identification number) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { roCuiChecksum } from '../../validators'

class RoCuiDetector extends RegexDetector {
  entityType = EntityType.RO_CUI
  score = 0.85
  contextKeywords = ['cui', 'cod unic', 'cod de inregistrare', 'fiscal code', 'cif']
  contextRequired = true

  pattern = /\b(?:RO)?\d{2,10}\b/gi
  validator = roCuiChecksum
}

registerRegion('ro', RoCuiDetector)

export { RoCuiDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/ro/cui.py
"""Romanian CUI (company identification number) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import ro_cui_checksum


@register_region("ro")
class CuiDetector(RegexDetector):
    entity_type = EntityType.RO_CUI
    score = 0.85
    context_keywords = [
        "cui", "cod unic", "cod de inregistrare", "fiscal code", "cif",
    ]
    context_required = True
    context_window = 50

    # Optional RO prefix + 2-10 digits
    pattern = re.compile(
        r"\b(?:RO)?\d{2,10}\b",
        re.IGNORECASE,
    )
    validator = staticmethod(ro_cui_checksum)
